cloudbridge/cloud/providers/gce/resources.py

```diff
@@ -162,3 +168,327 @@ def zones(self):
                  if zone['region'] == self._gce_region['selfLink']]
         return [GCEPlacementZone(self._provider, zone['name'], self.name)
                 for zone in zones]
+
+
+class GCEFirewallsDelegate(object):
+
+    _instance = None
+   
+    def __init__(self, provider):
+        if GCEFirewallsDelegate._instance is not None:
+            raise Exception('Use GCEFirewalls.instance() to instanciate.')
+        self._provider = provider
+        self._list_response = None
+        GCEFirewallsDelegate._instance = self
```

Comment:
It's not a singleton anymore. For some reason I was under the impression that there is going to be only one provider, ever.